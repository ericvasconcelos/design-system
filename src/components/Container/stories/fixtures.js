import styled from 'styled-components'

export const Content = styled.div`
  width: 100%;
  height: 100px;
  background: ${({ theme }) => theme.pallete.greyCashme[300]};
  display: flex;
  justify-content: center;
  align-items: center;
`
