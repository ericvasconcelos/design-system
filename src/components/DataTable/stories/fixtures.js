import { Button } from 'components/Button'
import { DropdownMenu } from 'components/DropdownMenu'

export const headers = [
  { key: 'header1', label: 'Header 1', sortable: false },
  { key: 'header2', label: 'Header 2', sortable: true },
  { key: 'header3', label: 'Header 3', sortable: true },
  { key: 'header4', label: 'Header 4', sortable: false }
]

export const data = [
  {
    id: 1,
    key: 1,
    header1: 'lorem ipsum',
    header2: 'R$ 1000',
    header3: 'lorem ipsum',
    header4: () => (
      <DropdownMenu
        buttonIcon="verticalMore"
        buttonLabel="Menu"
        buttonStyle="text"
        menuItems={[
          {
            key: 'item1',
            render: () => (
              <Button onClick={() => console.log('a')} styleType="secondary">
                Button 1
              </Button>
            )
          },
          {
            key: 'item2',
            render: () => (
              <Button onClick={() => console.log('b')} styleType="secondary">
                Button 2
              </Button>
            )
          }
        ]}
      />
    ),
    header5: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`
  },
  {
    id: 2,
    key: 2,
    header1: 'lorem ipsum',
    header2: 'R$ 1000',
    header3: 'lorem ipsum',
    header4: 'lorem ipsum',
    header5: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`
  },
  {
    id: 3,
    key: 3,
    header1: 'lorem ipsum',
    header2: 'R$ 1000',
    header3: 'lorem ipsum',
    header4: 'lorem ipsum',
    header5: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`
  },
  {
    id: 4,
    key: 4,
    header1: 'lorem ipsum',
    header2: 'R$ 1000',
    header3: 'lorem ipsum',
    header4: 'lorem ipsum',
    header5: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`
  },
  {
    id: 5,
    key: 5,
    header1: () => <p>Lorem ipsum function</p>,
    header2: 'R$ 1000',
    header3: 'lorem ipsum',
    header4: 'lorem ipsum',
    header5: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`
  }
]

export const collapse = [{ key: 'header5', label: 'Header 5' }]
