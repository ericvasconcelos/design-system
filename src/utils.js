export { arrayCombinations } from 'utils/arrayCombinations'
export { getBackgroundColor } from 'utils/getBackgroundColor'
export { getBorder } from 'utils/getBorder'
export { getButtonStyle } from 'utils/getButtonStyle'
export { getBorderRadius } from 'utils/getBorderRadius'
export { getColorFromName } from 'utils/getColorFromName'
export { getGuttersNames } from 'utils/getGuttersNames'
export { getIconNames } from 'utils/getIconNames'
export { getPadding } from 'utils/getPadding'
export { getPaddingNames } from 'utils/getPaddingNames'
export { getTypography } from 'utils/getTypography'
export { parsePalleteName } from 'utils/parsePalleteName'
export { styleColorValidator } from 'utils/styleColorValidator'
export { useAriaProps } from 'utils/useAriaProps'

export { DateUtils } from 'utils/DateUtils'
export { QueryUtils } from 'utils/QueryUtils'
export { ObjectUtils }  from 'utils/ObjectUtils'
export { CookiesUtils } from 'utils/CookiesUtils'
export { MoneyUtils } from 'utils/MoneyUtils'
export { NumberUtils } from 'utils/NumberUtils'