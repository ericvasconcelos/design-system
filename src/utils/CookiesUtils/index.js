export class CookiesUtils {
  static toObject(string = '') {
    const cookies = {}
    if (string !== '') {
      const parts = string.split('; ')
      parts.forEach(part => {
        const [key, value] = part.split('=')
        cookies[key] = value
      })
    }
    return cookies
  }
}
