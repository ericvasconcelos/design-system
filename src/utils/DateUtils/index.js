export class DateUtils {
  static toDate(string, format = 'pt-br') {
    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = string?.split('/')
        break
      case 'iso':
        [year, month, day] = string?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    return new Date(year, parseInt(month, 10) - 1, day)
  }

  static getAge(date) {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const difference = Date.now() - date.getTime()

    return Math.abs(new Date(difference).getUTCFullYear() - 1970)
  }

  static toString(date, format = 'pt-br') {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
    const month = date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1
    const year = date.getFullYear()

    switch (format) {
      case 'pt-br':
        return `${day}/${month}/${year}`
      case 'iso':
        return `${year}-${month}-${day}`
      default:
        throw new Error('Format invalid')
    }
  }

  static validate(date, format = 'pt-br') {
    const monthDaysMap = {
      '01': 31,
      '02': 29,
      '03': 31,
      '04': 30,
      '05': 31,
      '06': 30,
      '07': 31,
      '08': 31,
      '09': 30,
      10: 31,
      11: 30,
      12: 31
    }
    let isValid = false

    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = date?.split('/')
        break
      case 'iso':
        [year, month, day] = date?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    isValid = isValid || parseInt(day, 10) <= monthDaysMap[month]
    isValid = isValid && parseInt(year, 10) > 1900

    return isValid
  }

  static convertMonthToYear(month) {
    return parseInt(month, 10) / 12
  }

  static formatPeriod(period, periodType) {
    const periodTypeMap = {
      monthly: { singular: 'mÃªs', plural: 'meses' },
      yearly: { singular: 'ano', plural: 'anos' },
      daily: { singular: 'dia', plural: 'dias' }
    }
    return `${period} ${periodTypeMap[periodType][period < 2 ? 'singular' : 'plural']}`
  }
}
