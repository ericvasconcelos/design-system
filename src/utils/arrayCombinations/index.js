export const arrayCombinations = (...args) => {
  const returnable = []
  const max = args.length - 1
  function helper(arr, i) {
    for (let j = 0, { length } = args[i]; j < length; j++) {
      const newArray = arr.slice(0) // clone arr
      newArray.push(args[i][j])
      if (i === max) returnable.push(newArray)
      else helper(newArray, i + 1)
    }
  }
  helper([], 0)
  return returnable
}
