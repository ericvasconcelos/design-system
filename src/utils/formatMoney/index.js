import { toMoney, toNumber } from 'vanilla-masker'

export const formatMoney = entry => {
  const moneyFormatter = {
    precision: 2,
    separator: ',',
    delimiter: '.',
    unit: 'R$'
  }

  return toMoney(toNumber(entry), moneyFormatter)
}
