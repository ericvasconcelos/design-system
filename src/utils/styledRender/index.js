import React from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from 'styles/theme'

export const styledRender = (children, options = {}) => {
  const rendered = render(<ThemeProvider theme={theme}>{children}</ThemeProvider>, options)
  return {
    ...rendered,
    rerender: (ui, rerenderOptions = {}) =>
      styledRender(ui, { container: rendered.container, ...rerenderOptions })
  }
}
