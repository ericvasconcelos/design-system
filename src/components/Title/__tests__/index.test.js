import React from 'react'

import { styledRender } from 'utils/styledRender'

import { Title } from '../index.js'
import { fixtures } from './fixtures'

describe('should render the component correctly when', () => {
  it.each(fixtures)('props are %p', ({ children, level, styleType, styleColor, expectedCss }) => {
    const { getByTestId } = styledRender(
      <Title id="test" level={level} styleType={styleType} styleColor={styleColor}>
        {children}
      </Title>
    )
    const component = getByTestId('Title')

    expect(component).toHaveStyle(expectedCss)
    expect(component).toBeHtmlElementType(level || 'h1')
    expect(component).toMatchSnapshot()
  })
})
