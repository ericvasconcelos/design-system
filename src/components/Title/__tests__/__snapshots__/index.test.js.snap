// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.5rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "label", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.5rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.5rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "tag", "tag": "div"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.5rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.5rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "tag", "tag": "p"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.5rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.5rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "tag", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.5rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.5rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "tag", "tag": undefined} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.5rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.75rem;
font-weight: 500;
line-height: 1.5rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "button", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.75rem;
font-weight: 500;
line-height: 1.5rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "link", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.625rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "buttonSm", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.8125rem;
font-weight: 400;
line-height: 1.056rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "body3", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.8125rem;
  font-weight: 400;
  line-height: 1.056rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 0.9375rem;
font-weight: 400;
line-height: 1.219rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "body2", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 0.9375rem;
  font-weight: 400;
  line-height: 1.219rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 1.0625rem;
font-weight: 400;
line-height: 1.381rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "body1", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 1.0625rem;
  font-weight: 400;
  line-height: 1.381rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 1.0625rem;
font-weight: 400;
line-height: 1.381rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "input", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 1.0625rem;
  font-weight: 400;
  line-height: 1.381rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 1.0625rem;
font-weight: 500;
line-height: 1.381rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "body1Medium", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 1.0625rem;
  font-weight: 500;
  line-height: 1.381rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 1.5rem;
font-weight: 500;
line-height: 1.8rem;
text-decoration: none;
letter-spacing: -0.02rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "title3", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: -0.02rem;
  -moz-letter-spacing: -0.02rem;
  -ms-letter-spacing: -0.02rem;
  letter-spacing: -0.02rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 1.1875rem;
font-weight: 400;
line-height: 1.544rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "subtitle", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 1.1875rem;
  font-weight: 400;
  line-height: 1.544rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 2.25rem;
font-weight: 500;
line-height: 2.475rem;
text-decoration: none;
letter-spacing: -0.02rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "title2", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 2.25rem;
  font-weight: 500;
  line-height: 2.475rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: -0.02rem;
  -moz-letter-spacing: -0.02rem;
  -ms-letter-spacing: -0.02rem;
  letter-spacing: -0.02rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 3.625rem;
font-weight: 500;
line-height: 4rem;
text-decoration: none;
letter-spacing: -0.02rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "display", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 3.625rem;
  font-weight: 500;
  line-height: 4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: -0.02rem;
  -moz-letter-spacing: -0.02rem;
  -ms-letter-spacing: -0.02rem;
  letter-spacing: -0.02rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #003FF2;
font-size: 3rem;
font-weight: 500;
line-height: 3.3rem;
text-decoration: none;
letter-spacing: -0.02rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "lightBlueCash300", "styleType": "title1", "tag": "span"} 1`] = `
.c0 {
  color: #003FF2;
  font-size: 3rem;
  font-weight: 500;
  line-height: 3.3rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: -0.02rem;
  -moz-letter-spacing: -0.02rem;
  -ms-letter-spacing: -0.02rem;
  letter-spacing: -0.02rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;

exports[`should render the component correctly when props are {"children": "Ex ullamco aute est veniam sit.", "expectedCss": "color: #A6B5BB;
font-size: 0.5rem;
font-weight: 500;
line-height: 1rem;
text-decoration: none;
letter-spacing: 0rem;
font-family: 'DM Sans',sans-serif;
", "styleColor": "darkBlueCash100", "styleType": "tag", "tag": undefined} 1`] = `
.c0 {
  color: #A6B5BB;
  font-size: 0.5rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

<h1
  class="c0"
  data-testid="Title"
  id="test"
>
  Ex ullamco aute est veniam sit.
</h1>
`;
